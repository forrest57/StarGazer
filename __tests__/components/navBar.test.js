import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import NavBar from '../../components/navBar'



describe('<Navbar/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('correctly displays passed prop', async () => {
    render(<NavBar barText={'lorem'} />)
    const barText = screen.getByText('lorem')
    const gearIcon = await screen.getByTestId('GearButton')
    expect(barText).toBeTruthy()
    expect(gearIcon).toBeTruthy()
  })
  it('correctly conditionally hides the settings button', () => {
    render(<NavBar barText={'lorem'} isSettingsShown={false} />)
    let isExisting = true
    try {
      isExisting = screen.getByTestId('GearButton')
    } catch (_) {
      isExisting = false
    }
    expect(isExisting).toBe(false)
  })
})
