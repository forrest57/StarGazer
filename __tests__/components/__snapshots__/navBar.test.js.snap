// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 6,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 12,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 4,
        "textAlign": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      activeOpacity={1}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "display": "flex",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#374151",
            "borderRadius": 6,
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        testID="BackButton"
      >
        <Text />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 20,
      }
    }
    testID="barText"
  />
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      activeOpacity={1}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "display": "flex",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#374151",
            "borderRadius": 6,
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        testID="GearButton"
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
